src/web/components/Dropdown.tsx

```diff
@@ -157,132 +158,114 @@ const buttonExpanded = css`
-                document.removeEventListener('click', dismissOnClick);
-            };
-            const dismissOnClick = (event: MouseEvent) => {
-                event.stopPropagation();
-                this.setState(() => ({
-                    isExpanded: false,
-                }));
-                removeListeners();
-            };
-            const dismissOnEsc = (event: KeyboardEvent) => {
-                const escKey = 'Escape';
-
-                if (event.code === escKey) {
-                    this.setState(() => ({
-                        isExpanded: false,
-                    }));
-                    removeListeners();
-                }
-            };
-
-            document.addEventListener('keydown', dismissOnEsc, false);
-            document.addEventListener('click', dismissOnClick, false);
+        // Remove listener on unmount
+        return () => document.removeEventListener('click', dismissOnClick);
+    });
+
+    const dismissOnClick = (event: MouseEvent) => {
+        if (isExpanded) {
+            event.stopPropagation();
+            setIsExpanded(false);
         }
+    };
 
-        // needs to be unique to allow multiple dropdowns on same page
-        // this should be unique because JS is single-threaded
-        const dropdownID = `dropbox-id-${this.props.id}`;
-        const checkboxID = `checkbox-id-${this.props.id}`;
-
-        return (
-            <>
-                {this.state.noJS ? (
-                    <label
-                        htmlFor={checkboxID}
-                        className={cx({
-                            [button]: true,
-                            [buttonExpanded]: this.state.isExpanded,
-                        })}
-                        aria-controls={dropdownID}
-                        aria-expanded={this.state.isExpanded ? 'true' : 'false'}
-                        role="button"
-                    >
-                        <input
-                            className={input}
-                            type="checkbox"
-                            id={checkboxID}
-                            aria-controls={dropdownID}
-                            aria-checked="false"
-                            tabIndex={-1}
-                            key="OpenMainMenuCheckbox"
-                            role="menuitemcheckbox"
-                        />
-                        {label}
-                    </label>
-                ) : (
-                    <button
-                        onClick={this.boundToggle}
-                        className={cx({
-                            [button]: true,
-                            [buttonExpanded]: this.state.isExpanded,
-                        })}
+    const dismissOnEsc = (event: KeyboardEvent) => {
+        if (isExpanded && event.code === 'Escape') {
+            setIsExpanded(false);
+        }
+    };
+
+    const handleToggle = () => {
```

Comment:
Purely stylistic, but since this is a single statement what do you think about omitting the parenthesis?

Reply:
Done. Much cleaner.